server/src/main/java/org/opensearch/extensions/action/ExtensionActionRequest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions.action;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * This class enables Extension to make a transport request
+ *
+ * @opensearch.api
+ */
+public class ExtensionActionRequest extends ActionRequest {
+    private final String action;
+    private final byte[] requestBytes;
+
+    public ExtensionActionRequest(String action, byte[] requestBytes) {
```

Comment:
What "action" is expected as a `String` here?  A javadoc would help me understand.

Reply:
Sure thats a good point. 
"Action" is used in multiple ways so its confusing. But to simplify, any API is identified by an action defined by `ActionModule`[1]. Here, the extension will call the API it intends to invoke.

[1] https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/ActionModule.java

Reply:
Sounds like a great explanation to put in the `@param` tag of the javadoc!